xonsh/codecache.py

```diff
@@ -65,6 +65,7 @@ def run_compiled_code(code, glb, loc, mode):
     else:
         func = eval
     func(code, glb, loc)
+    builtins.__xonsh__.fstring_fields.clear()
```

Comment:
This is wrong! I probably need to stick the dictionary to the compiled block and let it get garbage-collected as the code block goes away.
```
$ Timer(1, lambda: print(f"{$HOME}")).start()
$ Exception in thread Thread-47:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 1254, in run
    self.function(*self.args, **self.kwargs)
  File "<xonsh-code>", line 1, in <lambda>
  File "/home/str/code/xonsh/xonsh/built_ins.py", line 1070, in eval_fstring_field
    field = fstring_fields.pop(field_id)
KeyError: 139889979616832
```